File: library/src/main/java/com/getstream/sdk/chat/adapter/AttachmentViewHolderFile.java
package com.getstream.sdk.chat.adapter;


import android.content.Context;
import android.graphics.drawable.Drawable;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.constraintlayout.widget.ConstraintLayout;

import com.getstream.sdk.chat.R;
import com.getstream.sdk.chat.utils.LlcMigrationUtils;
import com.getstream.sdk.chat.view.MessageListView;
import com.getstream.sdk.chat.view.MessageListViewStyle;

import io.getstream.chat.android.client.models.Attachment;
import io.getstream.chat.android.client.models.Message;

public class AttachmentViewHolderFile extends BaseAttachmentViewHolder {

    final String TAG = AttachmentViewHolder.class.getSimpleName();
    // Attachment
    private ConstraintLayout cl_attachment;
    private ImageView iv_file_thumb;
    private TextView tv_file_size, tv_file_title;

    private Context context;
    private MessageListItem.MessageItem messageListItem;
    private Message message;
    private Attachment attachment;
    private MessageListViewStyle style;
    private MessageListView.BubbleHelper bubbleHelper;
    private MessageListView.AttachmentClickListener clickListener;
    private MessageListView.MessageLongClickListener longClickListener;

    public AttachmentViewHolderFile(int resId, ViewGroup parent) {
        super(resId, parent);
        // Attach
        cl_attachment = itemView.findViewById(R.id.attachmentview);
        iv_file_thumb = itemView.findViewById(R.id.iv_file_thumb);
        tv_file_size = itemView.findViewById(R.id.tv_file_size);
        tv_file_title = itemView.findViewById(R.id.tv_file_title);
    }

    @Override
    public void bind(@NonNull Context context,
                     @NonNull MessageListItem.MessageItem messageListItem,
                     @NonNull Message message,
                     @NonNull AttachmentListItem attachmentItem,
                     @NonNull MessageListViewStyle style,
                     @NonNull MessageListView.BubbleHelper bubbleHelper,
                     @Nullable MessageListView.AttachmentClickListener clickListener,
                     @Nullable MessageListView.MessageLongClickListener longClickListener) {

        this.context = context;
        this.messageListItem = messageListItem;
        this.message = message;
        this.attachment = attachmentItem.getAttachment();
        this.style = style;
        this.bubbleHelper = bubbleHelper;
        this.clickListener = clickListener;
        this.longClickListener = longClickListener;

        applyStyle();
        configAttachment();
        configClickListeners();
    }

    private void applyStyle() {
        if(messageListItem.isMine()) {
            style.attachmentTitleTextMine.apply(tv_file_title);
            style.attachmentFileSizeTextMine.apply(tv_file_size);
        } else {
            style.attachmentTitleTextTheirs.apply(tv_file_title);
            style.attachmentFileSizeTextTheirs.apply(tv_file_size);
        }
    }

    private void configAttachment() {
        tv_file_size.setText(LlcMigrationUtils.getFileSizeHumanized(attachment));
        // update the icon nicely
        iv_file_thumb.setImageResource(LlcMigrationUtils.getIcon(attachment));
        tv_file_title.setText(attachment.getTitle());

        Drawable background = bubbleHelper.getDrawableForAttachment(messageListItem.getMessage(),
                messageListItem.isMine(),
                messageListItem.getPositions(),
                attachment);
        cl_attachment.setBackground(background);

    }

    private void configClickListeners(){
        cl_attachment.setOnClickListener(view -> {
            if (clickListener != null)
                clickListener.onAttachmentClick(message, attachment);
        });

        cl_attachment.setOnLongClickListener(view -> {
            if (longClickListener != null)
                longClickListener.onMessageLongClick(message);
            return true;
        });
    }
}

File: library/src/main/java/com/getstream/sdk/chat/adapter/viewholder/attachment/AttachmentViewHolderFile.kt
package com.getstream.sdk.chat.adapter.viewholder.attachment

import android.content.Context
import android.view.ViewGroup
import com.getstream.sdk.chat.adapter.AttachmentListItem
import com.getstream.sdk.chat.adapter.MessageListItem.MessageItem
import com.getstream.sdk.chat.adapter.inflater
import com.getstream.sdk.chat.databinding.StreamItemAttachmentFileBinding
import com.getstream.sdk.chat.utils.LlcMigrationUtils
import com.getstream.sdk.chat.view.MessageListView.AttachmentClickListener
import com.getstream.sdk.chat.view.MessageListView.BubbleHelper
import com.getstream.sdk.chat.view.MessageListView.MessageLongClickListener
import com.getstream.sdk.chat.view.MessageListViewStyle
import io.getstream.chat.android.client.models.Attachment
import io.getstream.chat.android.client.models.Message

class AttachmentViewHolderFile(
    parent: ViewGroup,
    private val binding: StreamItemAttachmentFileBinding =
        StreamItemAttachmentFileBinding.inflate(parent.inflater, parent, false)
) : BaseAttachmentViewHolder(binding.root) {

    private lateinit var messageListItem: MessageItem
    private lateinit var message: Message
    private lateinit var attachment: Attachment
    private lateinit var style: MessageListViewStyle
    private lateinit var bubbleHelper: BubbleHelper

    private var clickListener: AttachmentClickListener? = null
    private var longClickListener: MessageLongClickListener? = null

    override fun bind(
        context: Context,
        messageListItem: MessageItem,
        message: Message,
        attachmentListItem: AttachmentListItem,
        style: MessageListViewStyle,
        bubbleHelper: BubbleHelper,
        clickListener: AttachmentClickListener?,
        longClickListener: MessageLongClickListener?
    ) {
        this.messageListItem = messageListItem
        this.message = message
        this.style = style
        this.bubbleHelper = bubbleHelper

        this.clickListener = clickListener
        this.longClickListener = longClickListener

        attachment = attachmentListItem.attachment

        applyStyle()
        configAttachment()
        configClickListeners()
    }

    private fun applyStyle() {
        if (messageListItem.isMine) {
            style.attachmentTitleTextMine.apply(binding.tvFileTitle)
            style.attachmentFileSizeTextMine.apply(binding.tvFileSize)
        } else {
            style.attachmentTitleTextTheirs.apply(binding.tvFileTitle)
            style.attachmentFileSizeTextTheirs.apply(binding.tvFileSize)
        }
    }

    private fun configAttachment() {
        binding.tvFileSize.text = LlcMigrationUtils.getFileSizeHumanized(attachment)
        binding.ivFileThumb.setImageResource(LlcMigrationUtils.getIcon(attachment))
        binding.tvFileTitle.text = attachment.title

        val background = bubbleHelper.getDrawableForAttachment(
            messageListItem.message,
            messageListItem.isMine,
            messageListItem.positions,
            attachment
        )
        binding.attachmentview.background = background
    }

    private fun configClickListeners() {
        binding.attachmentview.setOnClickListener {
            clickListener?.onAttachmentClick(
                message,
                attachment
            )
        }
        binding.attachmentview.setOnLongClickListener {
            longClickListener?.onMessageLongClick(message)
            true
        }
    }
}
